contributing/webapp.md

```diff
@@ -32,146 +42,220 @@ This is a **straightforward guide** aims to setup all tools and settings on a Li
 - [Now what ?](#now-what--)
+    + [NCF api and Technique Editor](#ncf)
+    + [Entire Script](#full-script)
     + [Documentations](#documentations)
     + [Contribution](#contribution)
     + [Bug reports](#bug-reports)
     + [Community](#community)
 
-## Part 1 : Setup the local environment and the dev box
-Make sure you have installed [VirtualBox](https://www.virtualbox.org/wiki/Downloads) and Oracle VM VirutalBox Extension pack.
+## Part 0 : Installation of required packages and softwares
+
+### Needed packages
+If you are starting with a clean machine, you might want to take it all, should be needed at some point
+`apt update && apt install -y git openssh-server python3 python3-pip python python-pip openjdk-11-jdk net-tools ldap-utils maven`
+`pip3 install docopt requests pexpect urllib3` (or `pip`)
+
+Required only if you need to use the Technique Editor
+`apt update && apt install -y apache2` (could be `httpd` depending on your OS. RHEL for example)
+`pip3 install flask` (or `pip`)
+
+### VirtualBox installation
+```bash
+add-apt-repository "deb http://download.virtualbox.org/virtualbox/debian bionic contrib"
+wget -qO- https://www.virtualbox.org/download/oracle_vbox_2016.asc | sudo apt-key add -
+apt update
+vboxlatest=$(wget -qO- https://download.virtualbox.org/virtualbox/LATEST.TXT)
+vboxversion=$(echo $vboxlatest | cut -c1-3)
+apt install -y virtualbox-${vboxversion}
+wget https://download.virtualbox.org/virtualbox/${vboxlatest}/Oracle_VM_VirtualBox_Extension_Pack-${vboxlatest}.vbox-extpack
+echo y | sudo vboxmanage extpack install --replace Oracle_VM_VirtualBox_Extension_Pack-${vboxlatest}.vbox-extpack
+rm -rf Oracle_VM_VirtualBox_Extension_Pack-${vboxlatest}.vbox-extpack
+```
+
+### Vagrant installation
+```bash
+vagrantversion=$(wget -qO- https://raw.githubusercontent.com/hashicorp/vagrant/stable-website/version.txt) 
+wget https://releases.hashicorp.com/vagrant/${vagrantversion}/vagrant_${vagrantversion}_x86_64.deb
+apt install -y ./vagrant_${vagrantversion}_x86_64.deb
+rm -rf vagrant_${vagrantversion}_x86_64.deb
+```
+
+### Intellij Idea Community installation
+```bash
+wget https://download.jetbrains.com/idea/ideaIC-2019.3.3.tar.gz
+sudo tar -xzf ideaIC-2019.3.3.tar.gz -C /opt
+rm -rf ideaIC-2019.3.3.tar.gz
+```
+> For later use: `/opt/idea-IC-193.6494.35/bin/idea.sh` to start intellij
+
+### ldap apache directory studio installation
+```bash
+wget "http://apache.mirrors.ovh.net/ftp.apache.org/dist/directory/studio/2.0.0.v20180908-M14/ApacheDirectoryStudio-2.0.0.v20180908-M14-linux.gtk.x86_64.tar.gz"
+sudo tar -xzf ApacheDirectoryStudio-2.0.0.v20180908-M14-linux.gtk.x86_64.tar.gz -C /opt
+rm -rf ApacheDirectoryStudio-2.0.0.v20180908-M14-linux.gtk.x86_64.tar.gz
+```
+> For later use: `/opt/ApacheDirectoryStudio/ApacheDirectoryStudio` to start Apache Directory Studio
+
+### Elm installation
+```bash
+wget -qO elm.gz https://github.com/elm/compiler/releases/download/0.19.1/binary-for-linux-64-bit.gz
```

Comment:
Actually we use 0.19.0 not 0.19.1 and maybe give the link in case the installation change : https://guide.elm-lang.org/install/elm.html

Reply:
k updating